### ISSUE TITLE
Fix S2952 FP: Field disposal not recognized for some well known Dispose patterns

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6782]
Follow up to #6229

### Description

S2952 recognized this pattern:

```cs
 public void Cleanup() => fs.Dispose(); // Compliant: disposal is taking place in Dispose() too
 public void Dispose() => fs.Dispose();
```

These comparable constructs are not detected

1. Disposed in virtual Dispose(bool disposing) of the [dispose pattern](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern)

```cs
 public void Cleanup() => fs.Dispose(); // FP
 public void override Dispose(bool disposing) => fs.Dispose(); 
```

2. Disposed in [DisposeAsync](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync)

```cs
 public void Cleanup() => fs.Dispose(); // FP
 public Task DisposeAsync(bool disposing) => fs.Dispose(); 
```

### Repro steps

Please provide a minimal code snippet that compiles which reproduces the problem. Alternatively, you can share a link to the SonarCloud public project where the issue appears. If the issue requires some specific project configuration to reproduce, you can create a minimal reproducible project and share its git repository or attach it as a ZIP file.

### Expected behavior

Please provide a description of the behavior you expect.

### Actual behavior

Please provide a description of the actual behavior you observe.

### Known workarounds

Please provide a description of any known workarounds.

### Related information

* C#/VB.NET Plugins version
* Visual Studio version
* MSBuild / dotnet version
* SonarScanner for .NET version (if used)
* Operating System

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1846](https://sonarsource.atlassian.net/browse/NET-1846)

[NET-1846]: https://sonarsource.atlassian.net/browse/NET-1846?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
